gematria/datasets/pipelines/compile_modules_lib_test.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2024 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+import os
+
+from gematria.datasets.pipelines import compile_modules_lib
+from gematria.testing.python import ir_utils
+
+from apache_beam.testing.test_pipeline import TestPipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+import apache_beam as beam
+
+
+class CompileModulesTests(absltest.TestCase):
+
+  def test_optimize_modules(self):
+    ir_string = """
+define i32 @a() {
+  %a = add i32 3, 5
+  ret i32 %a
+}
+"""
+    module_optimizer = compile_modules_lib.OptimizeModules(
+        ['default<O0>', 'instcombine']
+    )
+    optimized_modules = list(
+        module_optimizer.process(ir_utils.get_bc_from_ir(ir_string))
+    )
+
+    self.assertLen(optimized_modules, 2)
+
+    optimized_ir_string = """
+define i32 @a() {
+  ret i32 8
+}
+"""
+    optimized_modules_text = [
+        ir_utils.get_ir_from_bc(module_bc) for module_bc in optimized_modules
+    ]
+
+    self.assertTrue(ir_string in optimized_modules_text[0])
+    self.assertTrue(optimized_ir_string in optimized_modules_text[1])
```

Comment:
`self.assertIn()` for cleaner code and better error messages.

Also, I don't think Beam makes any guarantees about the order of the values in the output. In this case, the order is preserved, but this is more of a coincidence (because the input is small, and there's just one worker running in one thread).

Reply:
In general yes, but we're explicitly calling the classes `process` function here, which will just map to the implementation in `compile_modules_lib.py` rather than going through any fancy Beam stuff where it might be reordered.